### PR TITLE
Add DecentralizedEnergyCommunity contract

### PR DESCRIPTION
**Pull request by hswopeams**
Add createCommunity function
Add addParticipantsToCommunity function

## Description

_Concise description of proposed changes, We recommend using screenshots and videos for better description_

## Additional Information

- [ ] I have read the [contributing docs](/scaffold-eth/scaffold-eth-2/blob/main/CONTRIBUTING.md) (if this is your first contribution)
- [ ] This is not a duplicate of any [existing pull request](https://github.com/scaffold-eth/scaffold-eth-2/pulls)

## Related Issues

_Closes #{issue number}_

_Note: If your changes are small and straightforward, you may skip the creation of an issue beforehand and remove this section. However, for medium-to-large changes, it is recommended to have an open issue for discussion and approval prior to submitting a pull request._

Your ENS/address:

### CODE REVIEW

File: packages/hardhat/contracts/DecentralizedEngergyCommunity.sol [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+/**
+ * @title   DecentralizedEnergyCommunity
+ * @notice  This contract is the main contract for the Decentralized Energy Community
+ */
+contract DecentralizedEnergyCommunity is AccessControl {
+	struct Community {
+		uint256 id;
```
Comment (borissmidt) [inline packages/hardhat/contracts/DecentralizedEngergyCommunity.sol]: i think uint32 is enough of uint20

**Review by DanyloPiatyhorets — DISMISSED** [review]

File: packages/hardhat/contracts/DecentralizedEngergyCommunity.sol [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		string name;
+		Participant[] participants;
+		bool active;
+		bool hasProducer;
+	}
+
+	struct Participant {
+		uint256 id;
```
Comment (borissmidt) [inline packages/hardhat/contracts/DecentralizedEngergyCommunity.sol]: there are not that many meters in belguim :P

File: packages/hardhat/contracts/DecentralizedEngergyCommunity.sol [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+ * @title   DecentralizedEnergyCommunity
+ * @notice  This contract is the main contract for the Decentralized Energy Community
+ */
+contract DecentralizedEnergyCommunity is AccessControl {
+	struct Community {
+		uint256 id;
+		string name;
+		Participant[] participants;
+		bool active;
+		bool hasProducer;
```
Comment (borissmidt) [inline packages/hardhat/contracts/DecentralizedEngergyCommunity.sol:14]: if want to optimize more 'deeply' we could try to use a byte and bitflip if they don't to this optimization for us.

File: packages/hardhat/contracts/DecentralizedEngergyCommunity.sol [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+	}
+
+	enum MeterType {
+		NONE,
+		PRODUCER,
+		CONSUMER
+	}
+
+	uint256 public communityIds;
+	uint256 public participantIds;
```
Comment (borissmidt) [inline packages/hardhat/contracts/DecentralizedEngergyCommunity.sol]: 16 bit might be enough?

File: packages/hardhat/contracts/DecentralizedEngergyCommunity.sol (hunk 3 above)
Comment (hswopeams) [reply packages/hardhat/contracts/DecentralizedEngergyCommunity.sol:14]: since id is now uint32, the bools can be packed into one slot with the id

**Review by DanyloPiatyhorets — APPROVED** [review]
types changed, all good

**Pull request merged**